src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -193,4 +193,43 @@ private CommandResult executeCommand(String commandText) throws CommandException
             throw e;
         }
     }
+
+    /**
+     * Handler for filling the command box when clicking the "Filter by Name" menu item.
+     */
+    @FXML
+    private void handleFilterByName() {
+        fillCommandBox("filter n/");
+    }
+
+    /**
+     * Handler for filling the command box when clicking the "Filter by Tag" menu item.
+     */
+    @FXML
+    private void handleFilterByTags() {
+        fillCommandBox("filter t/");
+    }
+
+    /**
+     * Handler for filling the command box when clicking the "Advanced Filter by Tag" menu item.
+     */
+    @FXML
+    private void handleAdvancedFilterByTags() {
+        fillCommandBox("advfilter t/[TAG] [OPERATOR] [VALUE]");
+    }
+
+    /**
+     * Heper method to fill the command box with a specified command text.
```

Comment:
The code in `MainWindow.java` is pretty well documented! It's easy to read and I can get a rough sensing of what's going on (even though I am not proficient in JavaFX).

Most of this file LGTM, there's just one small typo (`Heper` -> `Helper`) to fix on this line.

Reply:
A very major flaw, correction is imminent!